src/main/scala/io/scalac/tezos/translator/model/Contact.scala

```diff
@@ -0,0 +1,35 @@
+package io.scalac.tezos.translator.model
+
+sealed trait Contact extends Product with Serializable {
+  def getEmail(): String
+
+  def getPhone(): String
+}
+
+case class ContactPhone(number: String) extends Contact {
+  override def getEmail() = "Not declared"
+
+  override def getPhone() = number
+}
+
+case class ContactEmail(email: String) extends Contact {
+  override def getEmail() = email
+
+  override def getPhone() = "Not declared"
+}
+
+case class FullContact(phone: String, email: String) extends Contact {
+  override def getEmail() = email
+
+  override def getPhone() = phone
+}
+
+object Contact {
+  def tryToCreateContact(phone: String, email: String): Contact =
```

Comment:
we have [SendEmailDTOValidation ](https://github.com/ScalaConsultants/Tezos-FullStack-Console-backend/blob/dev/src/main/scala/io/scalac/tezos/translator/routes/dto/DTOValidation.scala#L68) where we check that email and phone not empty, the better approach here is just have something like `case class Contact(phone: String = "Not declared", email: String = "Not declared")`, also, you can use [refined types](https://github.com/fthomas/refined) to indicate explicitly that email and phone strings are not empty and season it with [new types](https://github.com/estatico/scala-newtype)

Reply:
Thanks for the advice but following the idea of @psssych I will change these stuff into more pater likable and remove this "Not declared" String.

Reply:
Don't worry, just suggestion, it's a good idea to change this stuff